[RPC] waitForThreadLocalRRefs returns jitFuture

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51696 [RPC] waitForThreadLocalRRefs returns jitFuture**
* #51695 [RPC] Move confirmation future in rref context to jit future

Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

rohan-varma committed:
[RPC] waitForThreadLocalRRefs returns jitFuture

Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] waitForThreadLocalRRefs returns jitFuture"


Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] waitForThreadLocalRRefs returns jitFuture"


Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] waitForThreadLocalRRefs returns jitFuture"


Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/rpc/rref_context.cpp:
```diff
@@ -653,26 +653,25 @@ void RRefContext::recordThreadLocalPendingRRefs() {
+  auto jitFuturePtr = c10::make_intrusive<JitFuture>(BoolType::get());
   if (userTable_.empty()) {
-    future = std::make_shared<Future<bool>>(true);
+    jitFuturePtr->markCompleted();
```
would I be correct if I assume this will mark it as completed using `IValue()` which by default is true?

mrshenli commented on torch/csrc/distributed/rpc/rref_context.cpp:
```diff
@@ -653,26 +653,25 @@ void RRefContext::recordThreadLocalPendingRRefs() {
         auto localCount = remainingRRefs->fetch_sub(1);
         if (localCount == 1) {
-          future->markCompleted(true);
+          jitFuturePtr->markCompleted(true);
```
any reason for passing true here but using empty arg above?

rohan-varma replied:
Oops, yeah will just use true for both. Although it doesn't matter too much, we don't consume the return value of this future, only care if it's completed or not

rohan-varma committed:
Update on "[RPC] waitForThreadLocalRRefs returns jitFuture"


Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] waitForThreadLocalRRefs returns jitFuture"


Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] waitForThreadLocalRRefs returns jitFuture"


Modify this API to use JitFuture.

Differential Revision: [D26239132](https://our.internmc.facebook.com/intern/diff/D26239132/)

[ghstack-poisoned]
